File: java18_tree_bst/tree_bst/src/p2.java

public class p2 {
    static class d{
      static void preorder(Node a){
            if(a == null){
                return;
            }
            System.out.print(a.val+" ");
          preorder(a.left);
          preorder(a.right);
        }
        static void inorder(Node a){
            if(a == null){
                return;
            }
            inorder(a.left);
            System.out.print(a.val+" ");

            inorder(a.right);
        }
        static void postorder(Node a){
            if(a == null){
                return;
            }

            postorder(a.left);
            postorder(a.right);
            System.out.print(a.val+" ");
        }
    }
    public static void main(String[] args) {
        Node a = new Node(1);

        Node b = new Node(2);
        Node c = new Node(3);

        Node d = new Node(4);
        Node e = new Node(5);
        Node f = new Node(6);
        Node g = new Node(7);

        a.right = c; a.left = b;
        b.right = e; b.left = d;
        c.right = g; c.left = f;


        //         1
        //       /   \
        //      2     3
        //    /  \   /  \
        //   4    5 6    7

        d kp = new d();

        kp.preorder(a);//1 2 4 5 3 6 7
        System.out.println();

        kp.inorder(a);//4 2 5 1 6 3 7
        System.out.println();

        kp.postorder(a);//4 5 2 6 7 3 1

    }
}

File: java20_tree_bst_p3/bst3/src/p2.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */

 // Range Sum of BST

function ans(root, low, high,arr){
  if(root == null) return ;
  if(root.val >=low && root.val <=high){
    arr.push(root.val);
  }
   if(root.val >=low){
     ans(root.left, low, high,arr)
   }
    if(root.val <=high){
    ans(root.right, low, high,arr)
   }
}

var rangeSumBST = function(root, low, high) {
    let arr = [0];
    if(root.left == null && root.right == null) {
      if(root.val >=low && root.val <=high) arr.push(root.val);
    }
    else ans(root, low, high,arr)
    return 0+ arr.reduce((add,el)=>add+el);
};